### ISSUE TITLE
Jinja in sls files not correctly executing the cmd.run command by powershell

### ISSUE THREAD

**Issue by konstest** [saltstack/salt#42716]
### Description of Issue/Question
Jinja in sls files not correctly executing the cmd.run command by powershell

### Setup
root@debugging-master:/srv/salt# cat cmd_state.sls 
{%- set rezult = salt['cmd.run']('[Environment]::GetEnvironmentVariable("Path", "Machine")', shell='powershell') %}
Run command by powershell:
  cmd.run:
    - name: | 
    {%- for str in rezult.split('\n') %}
        {{str}}
    {%- endfor %}
    - shell: powershell

### Steps to Reproduce Issue
Run cmd.run from CLI:
```
root@debugging-master:/srv/salt# salt windows-ci-03.build.ptsecurity.ru cmd.run '[Environment]::GetEnvironmentVariable("Path", "Machine")' shell=powershell
windows-ci-03.build.ptsecurity.ru:
    C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\salt;C:\Program Files\Docker;C:\Program Files (x86)\Git\cmd;C:\Program Files\doxygen\bin;C:\PYTHON34x64\;C:\PYTHON34x64\Scripts
```
From jinja in sls file:
```
root@debugging-master:/srv/salt# salt windows-ci-03.build.ptsecurity.ru state.show_sls cmd_state
windows-ci-03.build.ptsecurity.ru:
    ----------
    Run command by powershell:
        ----------
        __env__:
            base
        __sls__:
            cmd_state
        cmd:
            |_
              ----------
              name:
                  At line:1 char:43
                  + "[Environment]::GetEnvironmentVariable(\" Path\", \"Machine\")"
                  +                                           ~~~~~~~~~~~~~~~~~~~~~
                  Unexpected token 'Path\", \"Machine\")"' in expression or statement.
                      + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordEx 
                     ception
                      + FullyQualifiedErrorId : UnexpectedToken
            |_
              ----------
              shell:
                  powershell
            - run
            |_
              ----------
              order:
                  10000
```

### Versions Report
```
Salt Version:
           Salt: 2017.7.0
 
Dependency Versions:
           cffi: 1.5.2
       cherrypy: unknown
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: 0.24.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.24.0
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: ANSI_X3.4-1968
        machine: x86_64
        release: 4.4.0-79-generic
         system: Linux
        version: Ubuntu 16.04 xenial


Windows Minion: 2017.7
```

**Comment by konstest** [comment]
I tested this on salt-master 2016.11.6 (Carbon) - working all right

**Comment by bebehei** [comment]
I don't have the same environment, but I guess I have hit the same bug.

I'm using saltstack on Ubuntu 16.04 and 14.04 minions. The master uses Ubuntu 16.04. All master/minions are on version `2017.7.0 (Nitrogen)`.

As it seems, `cmd.run`'s argument won't get interpreted by shell. It will get called directly.

Creating a jinja-template with the following contents:
```
{{ salt['cmd.run']("echo asdf | tail -n 1") }}
```
Will yield to the output:
```
asdf | tail -n 1
```

Well, this rather shit than helpful.

> I tested this on salt-master 2016.11.6 (Carbon) - working all right

Before the update it worked fine, too.

**Comment by garethgreenaway** [comment]
@konstest Thanks for the report.  We'll get this one fixed.

**Comment by garethgreenaway** [comment]
@konstest Still working to figure out what changed between 2016.11 and 2017.7 that caused this to break, in the meantime can you try replacing cmd.run with cmd.shell to see if you see the same results?

**Comment by garethgreenaway** [comment]
FYI, found the issue.  previously for templates and state files cmd.run was aliased to cmd.shell because python_shell=True was needed to run commands that contained shell specific commands like `echo` or if things like pipes were used.  With the 2017.7 release this alias was removed, so those instances of cmd.run should specify python_shell=True or switch over to using cmd.shell.  Here is the warning that was included in the 2016.11 release:
```
[WARNING ] /usr/local/lib/python2.7/dist-packages/salt/utils/templates.py:73: DeprecationWarning: Starting in 2015.5, cmd.run uses python_shell=False by default, which doesn't support shellisms (pipes, env variables, etc). cmd.run is currently aliased to cmd.shell to prevent breakage. Please switch to cmd.shell or set python_shell=True to avoid breakage in the future, when this aliasing is removed.
```

And the release notes for 2017.7:
https://docs.saltstack.com/en/develop/topics/releases/2017.7.0.html

**Comment by konstest** [comment]
@garethgreenaway
Everything you suggested works. Thank you very much!

But i don't understand why when i running the same command from CLI all work without any changes:
cmd.run -> cmd.shell or adding python_shell=True 
??

And i don't was seeing this WARNING early:
```
[WARNING ] /usr/local/lib/python2.7/dist-packages/salt/utils/templates.py:73: DeprecationWarning: Starting in 2015.5, cmd.run uses python_shell=False by default, which doesn't support shellisms (pipes, env variables, etc). cmd.run is currently aliased to cmd.shell to prevent breakage. Please switch to cmd.shell or set python_shell=True to avoid breakage in the future, when this aliasing is removed.
```
